website/client/js/controllers/userCtrl.js

```diff
@@ -15,6 +15,13 @@ habitrpg.controller("UserCtrl", ['$rootScope', '$scope', '$location', 'User', '$
     $scope.$watch('_editing.profile', function(value){
       if(value === true) $scope.editingProfile = angular.copy(User.user.profile);
     });
+    
+    $scope.isBasicAnimal = function(name, type) {
+      var tmp = name.split('-');
+      $scope[type + "Egg"] = tmp[0];
+      $scope[type + "Potion"] = tmp[1];
+      return Content.hatchingPotions[$scope[type + "Potion"]] ? true : false;
+    }
```

Comment:
Put it in a relevant service.

Here's an example of similar code reuse in [partyCtrl.js](https://github.com/HabitRPG/habitrpg/blob/0880850408c1737598ee85bd88c6e9c26f872991/website/client/js/controllers/partyCtrl.js#L161), which has [questServices.js](https://github.com/HabitRPG/habitrpg/blob/develop/website/client/js/services/questServices.js#L4) injected as "Quests".